### PR TITLE
draft changes -- Bridget Torres -- similar_interest

### PR DESCRIPTION
**Pull request by bridgettorres**

### CODE REVIEW

File: pipeline/matching/generators/similar_interests.py [hunk 1]
```diff
@@ -1,10 +1,16 @@
+# imported itertools
```
Comment (erikqu) [inline pipeline/matching/generators/similar_interests.py:1]: Remove these misc. comments, only single line comment for clever logic (applies everywhere)

File: pipeline/matching/generators/similar_interests.py [hunk 2]
```diff
@@ -13,6 +19,43 @@ def __init__(self, min_common: int = 0):
...
+            ]  # list of interest of user two as a string
+            interest_one = set(list_of_interest_user_one)
+            interest_two = set(list_of_interest_user_two)
+            unique_interest = (interest_one).union(interest_two)  # all unique
+            common_interest = (interest_one).intersection(
+                interest_two
+            )  # to get common interest
+            # non zero division
+            calculate_score = 0.0
+            if len(unique_interest) != 0.0 and len(common_interest) != 0.0:
```
Comment (erikqu) [inline pipeline/matching/generators/similar_interests.py:44]: Length is never a float, lets check the length isn't `0`

File: pipeline/matching/generators/similar_interests_test.py [hunk 3]
```diff
@@ -41,8 +41,8 @@ def test_example():
         ),
     ]
     # TODO: Uncomment this assertion and delete the empty list assert
-    # assert actual == expected
-    assert actual == []
+    assert actual == expected
+    # assert actual == []
```
Comment (erikqu) [inline pipeline/matching/generators/similar_interests_test.py:47]: More tests for null attributes, users with no interests, users where only one has many interests etc

File: pipeline/matching/generators/similar_interests.py [hunk 4]
```diff
@@ -13,6 +19,43 @@ def __init__(self, min_common: int = 0):
...
+        for user_one, user_two in itertools.combinations(inp.users, r=2):
+            # union - consists of all the elements in interest of user one and user two without any repeats
+            # could use update but 'list' object has no attribute 'update'
+            list_of_interest_user_one = [
+                interests.name.lower() for interests in user_one.interests
+            ]  # list of interest of user one as a string
+            list_of_interest_user_two = [
+                interests.name.lower() for interests in user_two.interests
+            ]  # list of interest of user two as a string
+            interest_one = set(list_of_interest_user_one)
```
Comment (erikqu) [inline pipeline/matching/generators/similar_interests.py:36]: Maybe we should overwrite the old var name for consistency

**Review by erikqu — CHANGES_REQUESTED** [review]
Overall starting to look good!
